src/main/java/chatbot/tasks/DeadlineTask.java

```diff
@@ -0,0 +1,54 @@
+package chatbot.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+
+
+public class DeadlineTask extends Task {
+    private static String type = "D";
+    private LocalDate deadline;
+    private String deadlineString;
+
+    private void initDeadline(String deadline) {
+        if (deadline.matches("\\d{4}-\\d{2}-\\d{2}")) {
+            try {
+                this.deadline = LocalDate.parse(deadline);
+            } catch (DateTimeParseException e) {
+                this.deadlineString = deadline;
+            }
+        } else {
+            this.deadlineString = deadline;
+        }
+    }
+
+    public DeadlineTask(String name, String deadline) {
```

Comment:
Would it be more aligned with standard practice to define your constructors earlier in your code i.e. before the initDeadline(String deadline) function?

Reply:
yep noted, thanks!